router/core/wsproto/proto.go

```diff
@@ -0,0 +1,57 @@
+package wsproto
```

Comment:
Why did you re-implement this in the router? Is it not already implemented in graphql-go-tools?

Reply:
graphql-go-tools implements a client, while in the router we have to implement the server side. I could have taken the struct types and the message strings but I felt that graphql-go-tools should probably not export those at all. Additionally, the types in graphql-go-tools are named incorrectly in my opinion. Turns out the protocol named `graphql-ws` uses a websocket subprotocol called `graphql-transport-ws` while the protocol named `subscriptions-transport-ws` uses a websocket subprotocol named `graphql-ws`. The implementation in graphql-go-tools chose to use name the types after the websocket subprotocol, and I think using the official names is slightly less confusion.